contracts/modules/Payments.sol

```diff
@@ -36,6 +37,21 @@ abstract contract Payments is RouterImmutables {
         }
     }
 
+    /// @notice Approves a protocol to spend ERC20s in the router
+    /// @param token The token to approve
+    /// @param spenderID An ID signalling which protocol to approve
+    function approveERC20(address token, uint256 spenderID) internal {
+        // check spender is one of our approved spenders
+        address spender;
+        if (spenderID == 0) spender = SEAPORT;
+        else if (spenderID == 1) spender = SEAPORT_V2;
```

Comment:
An issue is that the router must approve the conduit address that the order is signed for - not the Seaport deployment addresses. We would use the OpenSea marketplace conduit address which we already have as a constant in our code. However, this can change in the future and we likely don't want to keep updating it. We should think about this more

Reply:
Resolved by discussing w/ seaport team , we are only approving to OS conduit